### PR TITLE
Initial 'not passed' QTS outcome flow added

### PR DESCRIPTION
**Pull request by jesseyuen**
Updates to this PR include:

## Alternate check your answer screens based on withdrawal status.

**Not passed, not withdrawing**

Screenshot_2020-11-03 Check QTS outcome details - Register trainee teachers - GOV UK(1)

**Not passed and withdrawing**

Screenshot_2020-11-03 Check QTS outcome details - Register trainee teachers - GOV UK

## Extended Flash banner

Screenshot_2020-11-04 Roderick Feil - Register trainee teachers - GOV UK

## Initial consideration for multiple QTS attempts, when the first attempt was not passed and the second attempt passed

**First attempt**

Screenshot 2020-11-04 at 10 50 29

**Second attempt**

Screenshot 2020-11-04 at 10 44 19

 ## Conditionally removing the space where the record-actions ui lives if it's not needed

**Before**

Screenshot 2020-11-09 at 09 25 31

**After

Screenshot 2020-11-04 at 10 47 00

### CODE REVIEW

File: app/routes/existing-record-routes.js [hunk 1]
```diff
@@ -76,9 +76,45 @@ module.exports = router => {
...
+  //     if (radioChoice == "Today") {
+  //       newRecord.qtsOutcomeRecordedDate = filters.toDateArray(filters.today())
+  //     }
+  //     if (radioChoice == "Yesterday") {
+  //       newRecord.qtsOutcomeRecordedDate = filters.toDateArray(moment().subtract(1, "days"))
+  //     } 
+  //     // res.redirect(`/record/${req.params.uuid}/reinstate/confirm`)
+  //   }
+  // })
+
```
Comment (edwardhorsford) [inline app/routes/existing-record-routes.js]: Do we need this?

File: app/routes/existing-record-routes.js [hunk 2]
```diff
@@ -76,9 +76,45 @@ module.exports = router => {
...
+    }
+    else {
+      let radioChoice = newRecord.qtsOutcomeRecordedDateRadio
+      if (radioChoice == "Today") {
+        newRecord.qtsOutcomeRecordedDate = filters.toDateArray(filters.today())
+      }
+      if (radioChoice == "Yesterday") {
+        newRecord.qtsOutcomeRecordedDate = filters.toDateArray(moment().subtract(1, "days"))
+      } 
+      // res.redirect(`/record/${req.params.uuid}/reinstate/confirm`)
```
Comment (edwardhorsford) [inline app/routes/existing-record-routes.js]: delete?

File: app/routes/existing-record-routes.js [hunk 3]
```diff
@@ -76,9 +76,45 @@ module.exports = router => {
...
+    }
+    
     if (_.get(data, "record.qtsDetails.standardsAssessedOutcome") == 'Not passed'){
       res.redirect(`/record/${req.params.uuid}/qts/not-passed/reason`)
     }
```
Comment (edwardhorsford) [inline app/routes/existing-record-routes.js]: Isn't a redirect needed where it's something other than not passed? or am I missing something?

File: app/views/_includes/forms/qts/outcome.html [hunk 4]
```diff
@@ -1,14 +1,36 @@
...
   {{ govukDateInput({
-    id: "assessment-end-date",
-    namePrefix: "record[programmeDetails][endDate]",
+    id: "custom-qtsOutcomeRecorded-date",
+    namePrefix: "record[qtsOutcomeRecordedDate]",
     fieldset: {
       legend: {
-        text: "On what date?",
-        classes: "govuk-fieldset__legend--ss"
+        text: "Enter the date the outcome was recorded",
```
Comment (edwardhorsford) [inline app/views/_includes/forms/qts/outcome.html]: I feel like this label could be simpler?

File: app/views/_includes/summary-cards/qts-assessment-not-passed-details.html [hunk 5]
```diff
@@ -24,6 +24,23 @@
       ]
     } if canAmend
   },
+  {
+    key: {
+      text: "Outcome recorded date"
```
Comment (edwardhorsford) [inline app/views/_includes/summary-cards/qts-assessment-not-passed-details.html]:
```suggestion
      text: "Outcome date"
```

**Review by edwardhorsford — COMMENTED** [review]
A couple minor nitpicks

File: app/routes/existing-record-routes.js (hunk 3 above)
Comment (jesseyuen) [reply app/routes/existing-record-routes.js]: Is this not what the second redirect is doing here?

```
if (_.get(data, "record.qtsDetails.standardsAssessedOutcome") == 'Not passed'){
  res.redirect(`/record/${req.params.uuid}/qts/not-passed/reason`)
}
else {
  res.redirect(`/record/${req.params.uuid}/qts/passed/confirm`)
}
```

File: app/routes/existing-record-routes.js (hunk 1 above)
Comment (jesseyuen) [reply app/routes/existing-record-routes.js]: Yes, but moved to another route.

**Review by edwardhorsford — APPROVED** [review]
woop woop! 👍

**Pull request merged**